agent/agent.go

```diff
@@ -466,24 +474,16 @@ func (r *agent) collectStatus(ctx context.Context) (systemStatus *pb.SystemStatu
 		}
 	}
 
-L:
 	for i := 0; i < len(members); i++ {
-		select {
-		case status := <-statusCh:
-			log.Debugf("Retrieved status from %v: %v.", status.member, status.NodeStatus)
-			nodeStatus := status.NodeStatus
-			if status.err != nil {
-				log.Warnf("Failed to query node %s(%v) status: %v.",
-					status.member.Name, status.member.Addr, status.err)
-				nodeStatus = unknownNodeStatus(&status.member)
-			}
-			systemStatus.Nodes = append(systemStatus.Nodes, nodeStatus)
-		case <-ctx.Done():
-			log.Warnf("Timed out collecting node statuses: %v.", ctx.Err())
-			// With insufficient status responses received, system status
-			// will be automatically degraded
-			break L
+		status := <-statusCh
```

Comment:
Here too, how is it going to timeout now if not received the status in time?

Reply:
I was thinking we don't want to timeout here. We should be waiting on and collecting all the node statuses. 
We are passing the context along the chain of functions required to gather status information. I thought as long as the final function call handles the timeout, all the functions higher up the chain should not block.

Reply:
Even if you don't want to exit immediately upon meeting the context deadline, it would be safer to use WaitGroup to make sure all goroutines are done before exit.
Also, this means updating `DialRPC` to accept the context as dialing a member atm is not restricted.